### PR TITLE
Updates for tests and other details

### PR DESCRIPTION
**Pull request by Prestaul**
- Added test cases
- Changed matching loop to search in reverse and perform better for smaller units (i.e. minutes matches faster than weeks)
- Updated docs with more formats
- Updated package.json with keywords and description

Needs a bump to v1.0.0 and a push to npm so npm has the readme!

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ module.exports = function(str) {
 	var i = 0, m;
 	str = str.toString();
 
-	for(; i < INTERVALS.length; i++) {
+	for(i = INTERVALS.length; i--;) {
```
Comment (nbroslawsky) [inline index.js]: You don't need a leading semicolon here?

Comment (nbroslawsky) [inline index.js]: Nevermind, I see what you did there. Sneaky.

Comment (Prestaul) [inline index.js]: I almost changed it to:

```js
var i = INTERVALS.length, m
str = str.toString();

while(i--) {
    // ...
}
```

Maybe that would have been clearer...

**Comment by selfcontained** [comment]
Nice, looks great to me.  You cool if we merge it @nbroslawsky ?  You'll probably have to `npm version major` it and `npm publish` though, I think you're the only registered collaborator on npm for it.

**Comment by nbroslawsky** [comment]
``` bash
$ npm publish
+ timestr@1.0.0
```

**Comment by Prestaul** [comment]
:+1: but also :cry:... I don't see it: https://www.npmjs.com/package/timestr

Any chance it is just slow... like over 5 minutes slow?

**Comment by Prestaul** [comment]
:boom: It's there! Thanks @nbroslawsky  and @selfcontained

**Comment by nbroslawsky** [comment]
image

**Comment by Prestaul** [comment]
Btw, I wanted to point Henrik to this today and that triggered the desire to get the readme updated on npm: https://twitter.com/Prestaul/status/588757940003164160

**Comment by nbroslawsky** [comment]
Nice!

**Pull request merged**